snuba/snuba_migrations/functions/0002_functions_v2.py

```diff
@@ -0,0 +1,266 @@
+    after: str
+
+
+UNKNOWN_SPAN_STATUS = 2
+
+new_columns: Sequence[NewColumn] = [
+    NewColumn(column=Column("function", String()), after="name"),
+    NewColumn(column=Column("module", String()), after="package"),
+    NewColumn(
+        column=Column("dist", String(Modifiers(nullable=True, low_cardinality=True))),
+        after="release",
+    ),
+    NewColumn(
+        column=Column("transaction_op", String(Modifiers(low_cardinality=True))),
+        after="dist",
+    ),
+    NewColumn(
+        column=Column(
+            "transaction_status", UInt(8, Modifiers(default=str(UNKNOWN_SPAN_STATUS)))
+        ),
+        after="transaction_op",
+    ),
+    NewColumn(
+        column=Column(
+            "http_method", String(Modifiers(nullable=True, low_cardinality=True))
+        ),
+        after="transaction_status",
+    ),
+    NewColumn(
+        column=Column(
+            "browser_name", String(Modifiers(nullable=True, low_cardinality=True))
+        ),
+        after="http_method",
+    ),
+    NewColumn(
+        column=Column("device_classification", UInt(8)),
+        after="browser_name",
+    ),
+]
+
+base_columns: List[Column] = [
+    Column("project_id", UInt(64)),
+    Column("transaction_name", String()),
+    Column("timestamp", DateTime()),
+    Column("fingerprint", UInt(64)),
+    Column("function", String()),
+    Column("package", String()),
+    Column("module", String()),
+    Column("is_application", UInt(8)),
+    Column("platform", String(Modifiers(low_cardinality=True))),
+    Column("environment", String(Modifiers(nullable=True, low_cardinality=True))),
+    Column("release", String(Modifiers(nullable=True, low_cardinality=True))),
+    Column("dist", String(Modifiers(nullable=True, low_cardinality=True))),
+    Column("transaction_op", String(Modifiers(low_cardinality=True))),
+    Column("transaction_status", UInt(8, Modifiers(default=str(UNKNOWN_SPAN_STATUS)))),
+    Column("http_method", String(Modifiers(nullable=True, low_cardinality=True))),
+    Column("browser_name", String(Modifiers(nullable=True, low_cardinality=True))),
+    Column("device_classification", UInt(8)),
+    Column("retention_days", UInt(16)),
+]
+
+materialized_columns: Sequence[Column] = base_columns + [
+    Column("count", AggregateFunction("count", [Float(64)])),
+    Column(
+        "percentiles",
+        AggregateFunction("quantiles(0.5, 0.75, 0.9, 0.95, 0.99)", [Float(64)]),
+    ),
+    Column("min", AggregateFunction("min", [Float(64)])),
+    Column("max", AggregateFunction("max", [Float(64)])),
+    Column("avg", AggregateFunction("avg", [Float(64)])),
+    Column("sum", AggregateFunction("sum", [Float(64)])),
+    Column("worst", AggregateFunction("argMax", [UUID(), Float(64)])),
+    Column("examples", AggregateFunction("groupUniqArray(5)", [UUID()])),
+]
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+
+    storage_set = StorageSetKey.FUNCTIONS
+
+    materialization_version = 1
```

Comment:
bumping the materialization version as we'll be changing the data that is sent to snuba as well